src/Data/Graph.purs

```diff
@@ -42,6 +45,9 @@ instance traversableGraph :: Traversable (Graph k) where
   traverse f (Graph m) = Graph <$> (traverse (\(v /\ ks) -> (_ /\ ks) <$> (f v)) m)
   sequence = traverse identity
 
+-- | An edge within a `Graph` referencing endpoint keys
+newtype Edge k = Edge (Tuple k k)
```

Comment:
Rather than defining an edge, it may be more helpful to define a newtype indicating which value inside the Tuple is the source and which is the destination. For example:
```diff
-newtype Edge k = Edge (Tuple k k)
+Tuple (Source k) (Destination k)
```
I'm not sure what the best API would be here.

Reply:
I've implemented this. I don't think it's any less ergonomic and, as you say, prevents mixing up source and destination. What do you think? Is this what you meant?